database/database.go

```diff
@@ -61,7 +61,7 @@ var (
 	mongoWriteConcern = "majority"
 	// mongoWriteConcernTimeout specifies a time limit, in milliseconds, for
 	// the write concern to be satisfied.
-	mongoWriteConcernTimeout = "1000"
+	mongoWriteConcernTimeout = "5000"
```

Comment:
you increased this to 5s, is it worth documenting why it's 5s? probably you ran into timeouts at 1?

Reply:
Well, we should definitely increase it to something more than 1s. I think 5s is probably a good value. It is indeed the value that you see appear in most examples, that's because it's the value used by the mongo documentation as well. 

I think this is a pretty dangerous setting for us due to our setup. Our replica set currently has 20 nodes, and the primary might be at the other side of the world right. I think this timeout is thrown when the primary doesn't acknowledge the write within the given timeout value. For us this means that we have to reach the primary, and a good chunk (majority) of the secondaries as well I believe, before it'll ack.

What I'm trying to say is that 5s might not give us enough leeway even. We should consider dropping it altogether or perhaps bump it to something like 30s? Which is still a reasonable request timeout value, and gives us even more leeway. It depends on how we treat the timeout, because the timeout doesn't mean the data is not persisted at the primary.

> A write operation that times out waiting for the specified write concern only indicates that the required number of replica set members did not acknowledge the write operation within the wtimeout time period. It does not necessarily indicate that the primary failed to apply the write. The data may exist on a subset of replica set nodes at the time of the write concern error, and can continue replicating until all nodes in the cluster have that data. Applications should take into account the potential availability of written data regardless of the state of write concern acknowledgment.

https://docs.mongodb.com/manual/core/replica-set-write-concern/

Reply:
I agree with that. I'm not sure I want to go for no limit at all because that will potentially block forever.

The 30s sounds very long but still reasonable, especially having in mind that `skyd` has similar timeouts.

I wouldn't really worry about data corruption because all of our operations are wrapped in transactions. A write concern error which aborts a given flow will cause the transaction to be aborted and rolled back.